agent/engine/containermetadata/types.go

```diff
@@ -0,0 +1,97 @@
+package containermetadata
+
+import (
+	"encoding/json"
+)
+
+// PortMapping holds data about the container's port bind to the host
+type PortMapping struct {
+	ContainerPort string
+	HostPort      string
+	BindIP        string
+	Protocol      string
+}
+
+// NetworkMetadata keeps track of the data we parse from the Network Settings
+// in docker containers
+type NetworkMetadata struct {
+	ports       []PortMapping
+	networkMode string
+	gateway     string
+	iPAddress   string
+	iPv6Gateway string
+}
+
+func (nm *NetworkMetadata) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Ports       []PortMapping `json:"PortMappings, omitempty"`
+		NetworkMode string        `json:"NetworkMode, omitempty"`
+		Gateway     string        `json:"Gateway, omitempty"`
+		IPAddress   string        `json:"IPAdress, omitempty"`
+		IPv6Gateway string        `json:"IPv6Gateway, omitempty"`
+	}{
+		Ports:       nm.ports,
+		NetworkMode: nm.networkMode,
+		Gateway:     nm.gateway,
+		IPAddress:   nm.iPAddress,
+		IPv6Gateway: nm.iPv6Gateway,
+	})
+}
+
+// DockerContainerMetadata keeps track of all metadata acquired from Docker inspection
+// Has redundancies with engine.DockerContainerMetadata but packages all
+// docker metadata we want in the service so we can change features easily
+type DockerContainerMetadata struct {
+	status        string
+	containerID   string
+	containerName string
+	imageID       string
+	imageName     string
+	networkInfo   *NetworkMetadata
+}
+
+// TaskStaticMetadata keeps track of all metadata associated with a task
+// provided by AWS, does not depend on the creation of the container
+type TaskStaticMetadata struct {
```

Comment:
I'm not in favor of having a name include `Static`. 
Could you please change this ?